src/java/org/apache/cassandra/tools/NodeTool.java

```diff
@@ -375,7 +387,11 @@ public void runInternal()
 
             try (NodeProbe probe = connect())
             {
-                execute(probe);
+                if (nodeTool != null)
```

Comment:
there are tools like Sjk which are executing this differently and not everytime nodetool is present.